services/registry/handlers/maintainers.js

```diff
@@ -16,88 +10,70 @@ module.exports = [
   ),
   fork.post(
     '/v1/packages/package/:namespace([^@]+)@:host/:name/maintainers/:invitee',
-    findInvitee(canWrite(invite))
+    authn.required(invite)
   ),
   fork.del(
     '/v1/packages/package/:namespace([^@]+)@:host/:name/maintainers/:invitee',
-    findInvitee(canWrite(remove))
-  ),
-  fork.post(
-    '/v1/packages/package/:namespace([^@]+)@:host/:name/invitation/:member',
-    packageExists(isNamespaceMember(accept))
-  ),
-  fork.del(
-    '/v1/packages/package/:namespace([^@]+)@:host/:name/invitation/:member',
-    packageExists(isNamespaceMember(decline))
+    authn.required(remove)
```

Comment:
A theme emerges: registry endpoints can make authorization decisions on a large scale – "this will fail if you aren't authenticated" – but delegate granular authorization decisions to the storage layer ("can Alice write to Bob's package?")